src/app/InteractionModelEngine.h

```diff
@@ -408,6 +408,13 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
     InteractionModel::DataModel * GetDataModel() const;
 
+    // MUST NOT be used while the interaction model engine is running as interaction
```

Comment:
I am looking at it ... but for example `Init` claims we have a kState_NotInitialized however there is no such thing in practice (yay for refering to private details when describing return errors...).

So unsure if there is a proper way to do this. We could check if Init was called by checking some members, but comments claim that #21233 tracks that we cannot have clean shutdowns :(

Will spend some time on it, but I may not be able to find something reasonable given that IM lifetime seems odd overall.

Reply:
For now added a check of `read handlers are empty` since read handlers are the ones that use iteration in general and should have some state.

Not ideal, but better than nothing.